python/pyiceberg/catalog/rest.py

```diff
@@ -0,0 +1,371 @@
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+from typing import (
+    Dict,
+    List,
+    Optional,
+    Set,
+    Tuple,
+    Type,
+    Union,
+)
+
+import requests
+from pydantic import Field
+from requests import HTTPError
+
+from pyiceberg.catalog import Identifier, Properties
+from pyiceberg.catalog.base import Catalog, PropertiesUpdateSummary
+from pyiceberg.exceptions import (
+    AlreadyExistsError,
+    BadCredentialsError,
+    BadRequestError,
+    NoSuchNamespaceError,
+    NoSuchTableError,
+    RESTError,
+    ServerError,
+    ServiceUnavailableError,
+    TableAlreadyExistsError,
+    UnauthorizedError,
+)
+from pyiceberg.schema import Schema
+from pyiceberg.table.base import Table
+from pyiceberg.table.partitioning import UNPARTITIONED_PARTITION_SPEC, PartitionSpec
+from pyiceberg.table.sorting import UNSORTED_SORT_ORDER, SortOrder
+from pyiceberg.utils.iceberg_base_model import IcebergBaseModel
+
+
+class Endpoints:
+    get_config: str = "config"
+    list_namespaces: str = "namespaces"
+    create_namespace: str = "namespaces"
+    load_namespace_metadata: str = "namespaces/{namespace}"
+    drop_namespace: str = "namespaces/{namespace}"
+    update_properties: str = "namespaces/{namespace}/properties"
+    list_tables: str = "namespaces/{namespace}/tables"
+    create_table: str = "namespaces/{namespace}/tables"
+    load_table: str = "namespaces/{namespace}/tables/{table}"
+    update_table: str = "namespaces/{namespace}/tables/{table}"
+    drop_table: str = "namespaces/{namespace}/tables/{table}?purgeRequested={purge}"
+    table_exists: str = "namespaces/{namespace}/tables/{table}"
+    get_token: str = "oauth/tokens"
+    rename_table: str = "tables/rename"
+
+
+AUTHORIZATION_HEADER = "Authorization"
+BEARER_PREFIX = "Bearer"
+CATALOG_SCOPE = "catalog"
+CLIENT_ID = "client_id"
+PREFIX = "prefix"
+CLIENT_SECRET = "client_secret"
+CLIENT_CREDENTIALS = "client_credentials"
+CREDENTIAL = "credential"
+GRANT_TYPE = "grant_type"
+SCOPE = "scope"
+TOKEN_EXCHANGE = "urn:ietf:params:oauth:grant-type:token-exchange"
+
+NAMESPACE_SEPARATOR = b"\x1F".decode("UTF-8")
+
+
+class CreateTableRequest(IcebergBaseModel):
+    name: str = Field()
+    location: Optional[str] = Field()
+    table_schema: Schema = Field(alias="schema")
+    partition_spec: Optional[PartitionSpec] = Field(alias="partition-spec")
+    write_order: Optional[SortOrder] = Field(alias="write-order")
+    stage_creation: bool = Field(alias="stage-creation", default=False)
+    properties: Dict[str, str] = Field(default_factory=dict)
+
+
+class TokenResponse(IcebergBaseModel):
+    access_token: str = Field()
+    token_type: str = Field()
+    expires_in: int = Field()
+    warehouse_id: str = Field()
```

Comment:
`region` and `warehouse_id` aren't in the spec. Maybe this is something returned by an implementation? We probably don't want to include them here.

Reply:
Looks like it is returned by an implementation ;)